app/Http/Controllers/ContestsController.php

```diff
@@ -16,6 +16,11 @@ public function index()
     {
         $contests = Contest::all();
 
+        foreach ($contests as $contest) {
+            $contest->signup_end_date = $contest->waitingRoom->signup_end_date;
+            $contest->signups_open = ($contest->waitingRoom->isOpen()) ? 'Open' : 'Closed';
```

Comment:
not a blocker, but this could technically be done in the template (so we don't need to do a second loop). not a big deal, just an idea